CRM/Admin/Form/Setting/Search.php

```diff
@@ -69,6 +70,10 @@ public function buildQuickForm() {
     // Autocomplete for Contact Reference (custom fields)
     $element = $this->getElement('contact_reference_options');
     $element->_elements[0]->_flagFrozen = TRUE;
+
+    // Freeze first element of quicksearch options
+    $element = $this->getElement('quicksearch_options');
```

Comment:
this should not be needed

Reply:
& if it is then we need to figure out a metadata way

Reply:
I added it because the first checkbox should not be unselected.

Reply:
OK - we should think about a way to denote that in the setting metadata - what is it & why can't it be deselected?

Reply:
See screenshot.

Reply:
Hmm - perhaps we add a key to the metadata for 'required_options' or similar

Reply:
(see discussion on chat about how we imagine removing that form.....)

Reply:
Actually I wonder if that field per se even IS required - if the field itself were mandatory & people had to enter 'something' that seems fine

Reply:
Yea, it would be pretty weird to not allow searching for contacts by name, but technically, yea it's possible.

Reply:
It might be better to get a metadata 'required_options' since the search form is a reasonable target for elimination & replacement by a Generic form with url params & the ONLY thing on that form that causes it to exist is the 3 freeze statements.

(the 3 functions should go to select values & the url param will remove the need for the settings array once we have done that last piece)

Reply:
Hmm - I guess there are already similar on the same form so I could wear that - in general my feeling is that we shouldn't permit anything else that is not meta-data driven onto any settings forms any more